### ISSUE TITLE
Thread Error while connecting

### ISSUE THREAD

**Issue by Zacktamondo** [LenhartStephan/flutter_blue_classic#2]
When trying to connect to a device using the example app if I waited some time, I get the following error:
`Thread[3,tid=32754,WaitingInMainSignalCatcherLoop,Thread*=0x7a32e39800,peer=0x138c0150,"Signal Catcher"]: reacting to signal 3`
and a Force close or wait dialog on device.
As I read for a little bit, the cause is using a thread on the main UI thread.
Is there any info I forgot to mention? 

Thanks a lot for the package by the way.

**Comment by LenhartStephan** [comment]
Could you describe the issue in a bit more detail? What did you do? (Scan for devices, then wait (how long), then connect?) 
Also could you provide a little more info on the devices involved (Android-Version, Manufacturer, ...) and if you have, a more detailed crash log.

**Comment by Zacktamondo** [comment]
**Steps to reproduce:**

1. Forget the device if previously paired.
2. Scan for devices.
3. Select device.
4. When pair auth dialog appears, wait for 30 seconds and the force close dialog appears.

**My device info:**
Android 14 (Custom ROM)
Xiaomi Poco F1

**Full crash log:**
```
Launching lib/main.dart on POCO F1 in debug mode...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:39407/WK6eLH8TXEY=/ws
I/Gralloc4(24117): mapper 4.x is not supported
W/Gralloc3(24117): mapper 3.x is not supported
I/Gralloc2(24117): Adding additional valid usage bits: 0x8202000
D/CompatibilityChangeReporter(24117): Compat change id reported: 263076149; UID 10525; state: DISABLED
D/FlutterBlueClassic(24117): Connecting to 14:6A:0D:78:90:DF (id: 1)
I/classic_example(24117): Thread[5,tid=24122,WaitingInMainSignalCatcherLoop,Thread*=0x7d6236c400,peer=0x12c801a0,"Signal Catcher"]: reacting to signal 3
I/classic_example(24117): 
I/classic_example(24117): Wrote stack traces to tombstoned
I/flutter (24117): PlatformException(connect_error, read failed, socket might closed or timeout, read ret: -1, null, null)
```

**Comment by LenhartStephan** [comment]
Thanks for the description and logs. I was able to reproduce the error and will provide the fixes with the next update 0.0.3.

**Comment by Zacktamondo** [comment]
I am happy that I could help, and thanks a lot for your work.
